rclpy/rclpy/utilities.py

```diff
@@ -71,9 +72,37 @@ def get_rmw_implementation_identifier():
 
 
 def get_available_rmw_implementations():
-    """Return the set of all available RMW implementations as registered in the ament index."""
-    rmw_implementations = ament_index_python.get_resources('rmw_typesupport')
-    return {name for name in rmw_implementations if name != 'rmw_implementation'}
+    """
+    Return the set of all available RMW implementations as registered in the ament index.
+
+    The result can be overridden by setting an environment variable named
+    ``RMW_IMPLEMENTATIONS``.
+    The variable can contain RMW implementation names separated by the platform
+    specific path separator.
+    Including an unavailable RMW implementation results in a RuntimeError.
+    """
+    available_rmw_implementations = ament_index_python.get_resources(
+        'rmw_typesupport')
+    available_rmw_implementations = {
+        name for name in available_rmw_implementations
+        if name != 'rmw_implementation'}
+
+    # filter by implementations in environment variable if provided
+    rmw_implementations = os.environ.get('RMW_IMPLEMENTATIONS')
+    if rmw_implementations:
+        rmw_implementations = rmw_implementations.split(os.pathsep)
+        missing_rmw_implementations = set(rmw_implementations) - \
```

Comment:
style nit: [pep8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) prefers wrapping line continuation in parentheses instead of using a backslash.

Reply:
Since we use the backslash style throughout our code I am less inclined to diverge from that style here.

Reply:
I don't know if it's that prevalent; there's a dozen uses in rclpy that I'm inclined to go and change :grin:

Reply:
`pep8` calls the parenthesis a preference only and the current code also passes `flake8` just fine.